pkg/tfbridge/detailed_diff.go

```diff
@@ -260,6 +297,78 @@ func (differ detailedDiffer) makeListDiff(
 	return diff
 }
 
+func (differ detailedDiffer) makeSetDiff(
+	path propertyPath, old, new resource.PropertyValue,
+) map[detailedDiffKey]*pulumirpc.PropertyDiff {
+	type setChangeIndex struct {
+		engineIndex   int
+		newStateIndex int
+		oldChanged    bool
+		newChanged    bool
+	}
+
+	diff := make(map[detailedDiffKey]*pulumirpc.PropertyDiff)
+	oldList := old.ArrayValue()
+	newList := new.ArrayValue()
+	newInputsList := []resource.PropertyValue{}
+
+	newInputs, newInputsOk := path.GetFromMap(differ.newInputs)
+	if newInputsOk && isPresent(newInputs) && newInputs.IsArray() {
+		newInputsList = newInputs.ArrayValue()
+	}
+
+	oldIdentities := differ.calculateSetHashIndexMap(path, oldList)
+	newIdentities := differ.calculateSetHashIndexMap(path, newList)
+	inputIdentities := hashIndexMap{}
+
+	if !pathContainsComputed(path, differ.tfs, differ.ps) {
+		// The inputs are only safe to hash if the schema has no computed properties
+		inputIdentities = differ.calculateSetHashIndexMap(path, newInputsList)
+	}
+
+	// The old indices and new inputs are the indices the engine can reference
+	// The new state indices need to be translated to new input indices when presenting the diff
+	setIndices := make(map[int]setChangeIndex)
+	for hash, oldIndex := range oldIdentities {
+		if _, newOk := newIdentities[hash]; !newOk {
+			setIndices[oldIndex] = setChangeIndex{engineIndex: oldIndex, oldChanged: true, newStateIndex: -1, newChanged: false}
+		}
+	}
+	for hash, newIndex := range newIdentities {
+		if _, oldOk := oldIdentities[hash]; !oldOk {
+			inputIndex := inputIdentities[hash]
+			if inputIndex == -1 {
+				GetLogger(differ.ctx).Warn(fmt.Sprintf(
+					"Element at path %s in new state not found in inputs, the displayed diff might be inaccurate",
```

Comment:
Think of UX on this message. Perhaps something like "Additional changes detected: ".